src/targets/anvil.js

```diff
@@ -0,0 +1,104 @@
+import { ClickableObject } from "../clickable-object.js";
+import { CONSTANTS } from "../constants/constants.js";
+import { characterData } from "../cookie-io.js";
+import { Button } from "../ui/button.js";
+import { getItemClass, hasItem } from "../utilities.js";
+
+export class Anvil extends ClickableObject {
+    name = "Anvil";
+    varName = "bronzeDagger"; // Use # bronze daggers forged as quest
+    examineText = "Used for fashioning metal items.";
+    actions = [
+        { text: "Forge", func: "clickTarget" },
+        { text: "Examine", func: "examine" },
+    ];
+
+    constructor(scene) {
+        super();
+        this.scene = scene;
+
+        const cameraWidth = scene.cameras.main.width;
+        const cameraHeight = scene.cameras.main.height;
+        const width = 220;
+        const height = 140;
+        const x = cameraWidth / 2 - width + 60;
+        const y = cameraHeight / 2 - height + 40;
+
+        // Add invisible button for anvil
+        this.sprite = new Button(scene, x, y, width, height);
+        this.sprite.on("pointerdown", (pointer) => {
+            if (pointer.rightButtonReleased() && !pointer.leftButtonDown()) {
+                this.createRightClickMenu(pointer.x, pointer.y, this.actions);
+            } else {
+                this.clickTarget();
+            }
+        });
+    }
+
+    async clickTarget() {
+        const inv = this.scene.dashboard.inventory;
+        const chat = this.scene.scene.get(CONSTANTS.SCENES.CHAT);
+
+        // See if bar is selected
+        const selectedIndex = inv.curSelectedItemIndex;
+        if (selectedIndex < 0) {
+            chat.writeText("Select a bar in your inventory first.");
+            return;
+        }
+
+        const hasHammer = await hasItem(inv.inventory, "Hammer");
+
+        if (hasHammer === false) {
+            chat.writeText("You need a hammer to work the metal with.");
+            return;
+        }
+
+        const selectedItem = inv.inventory[selectedIndex];
+
+        switch (selectedItem.name) {
+            case "Bronze Bar":
+                this.smith("BronzeDagger");
```

Comment:
Just putting a note here for our future selves when we have more smithing recipes - Setting the smithing recipe should be done by right clicking the anvil so that it doesn't have to be chosen on every click. This will probably change to this.smith("BronzeItem") and the smith method will check the anvil state.

Reply:
I added a card to Trello under planned features as a reminder, since this is not within my planned scope for this pull.